### PR TITLE
[#1][#2] Generate project from rails-templates

### LINKED ISSUE #1: Generate project from rails-templates
## Why

Generating a project from our templates will give us the essential core to start off conveniently.

## Acceptance Criteria

Generate a project from rails-templates

Also, update the following files accordingly (after the generation):

- `CODEOWNERS`  → Add Toby, Huey, Olivier & Hoang
- `README.md` → Update `rails-templates` to `ic-rails-huey-toby`

## Design

N/A

## Resources

https://github.com/nimblehq/rails-templates

### LINKED ISSUE #2: Set up CI to execute unit tests
## Why

In order to avoid any code smells, while maintaining code quality, a proper CI must be added.

**Note**: This has already been added in `test.yml`, but we must verify it's actually working correctly

## Acceptance Criteria

Every time a pull request is created or updated, the _unit_ tests must be run automatically (in GitHub Actions).

## Design

N/A

## Resources

N/A

### PR DESCRIPTION
**Pull request by hoangnguyen92dn**
- Close #1
- Close #2

## What happened 👀

Generate the project from [rails-templates](https://github.com/nimblehq/rails-templates)

## Insight 📝

- Add Github secrets to publish the wiki
- Update PR template
- Update deployment workflow branches:
  - Rename `development` -> `develop`
  - Remove `master` branch
- The `coverage` folder doesn't generate to collect the coverage. Here is the fix [0ca7278](https://github.com/nimblehq/ic-rails-huey-toby/pull/5/commits/0ca72789b6684ac8fea70cabb43215fe23ff5332) 
- Execute unit tests failed to generate `coverage.json` file so I changed the command to get rid of `--exclude-pattern`: [c14b913](https://github.com/nimblehq/ic-rails-huey-toby/pull/5/commits/c14b913765c44a5b2f2224bd36edc8ecfd55e324). Any other advise to bypass this step are welcome 🙏 

## Proof Of Work 📹

- Start Docker
- Execute: `make dev`

image

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.DOCKER_REGISTRY_HOST }}
+          username: ${{ env.DOCKER_REGISTRY_USERNAME }}
+          password: ${{ env.DOCKER_REGISTRY_TOKEN }}
+
+      - name: Pull Docker image
+        run: docker-compose pull test || true
+
+      - name: Run unit tests
```
Comment (olivierobert) [inline .github/workflows/test.yml:100]: That can besimplified to run all tests:

```suggestion
      - name: Run tests
```

**Review by olivierobert — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        run: docker-compose run test bundle exec rspec --profile --format progress
+
+      - name: Upload tests coverage artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: coverage
+          path: coverage
+
+  system_tests:
```
[the remark this answers is not in the document]
Comment (hoangnguyen92dn) [reply .github/workflows/test.yml:109]: @hoangmirs It's understandable if we combine the step to reduce the time of pulling the image. But I think we should keep it separated for each step, it's cleaner and easier to follow up on each step. cc @Tuubz

File: .github/wiki/Getting-Started.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+### Docker
+
+- Install [Docker for Mac](https://docs.docker.com/docker-for-mac/install/)
+
+### Development
+
+Run the Rails app:
+
+```sh
+make dev
```
[the remark this answers is not in the document]
Comment (hoangnguyen92dn) [reply .github/wiki/Getting-Started.md:14]: @hoangmirs I think we could consider to update the `rails-template` 👍

File: .github/workflows/test.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (hoangnguyen92dn) [reply .github/workflows/test.yml]: Updated [c928293](https://github.com/nimblehq/ic-rails-huey-toby/pull/5/commits/c928293781b6d60a65ecf3e766405d0c680a5777)

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Team
+# @Tuubz is the Team Lead
+* @hoangnguyen92dn @hoangmirs @olivierobert
```
[the remark this answers is not in the document]
Comment (hoangnguyen92dn) [reply .github/CODEOWNERS]: Updated [9f3a0d9](https://github.com/nimblehq/ic-rails-huey-toby/pull/5/commits/9f3a0d9ee690cb9607956ff94f8fc95759c2ce16)

File: .github/PULL_REQUEST_TEMPLATE/RELEASE_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+Link to the milestone on Github e.g. https://github.com/nimblehq/ic-rails-huey-toby/milestone/41?closed=1  
+or link to the release on the project management tool e.g. https://app.shortcut.com/nimblehq/label/12345
```
[the remark this answers is not in the document]
Comment (hoangnguyen92dn) [reply .github/PULL_REQUEST_TEMPLATE/RELEASE_TEMPLATE.md]: Updated [39bc5ca](https://github.com/nimblehq/ic-rails-huey-toby/pull/5/commits/39bc5ca8794f63314c9c40960b28c4e83a6bd6ce)

File: .github/wiki/Getting-Started.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (hoangnguyen92dn) [reply .github/wiki/Getting-Started.md:20]: Updated [b81f24d](https://github.com/nimblehq/ic-rails-huey-toby/pull/5/commits/b81f24dfe9c316135a7ac6d7a695cf6ab1b2c268)

File: .github/workflows/test_production_build.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
+name: Test production build
+
+on:
+  push:
+    branches-ignore:
+      - master
```
[the remark this answers is not in the document]
Comment (hoangnguyen92dn) [reply .github/workflows/test_production_build.yml]: Updated [01515c7](https://github.com/nimblehq/ic-rails-huey-toby/pull/5/commits/01515c7f63cc073ae64d37ecd8f05210f3317305)

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+env/setup:
+	./bin/envsetup.sh
+	rails db:prepare
+
+env/teardown:  # this command will delete data
+	./bin/envteardown.sh
+
+install-dependencies:
+	bundle install
+	yarn install
```
[the remark this answers is not in the document]
Comment (hoangnguyen92dn) [reply Makefile]: Updated [c0e5c22](https://github.com/nimblehq/ic-rails-huey-toby/pull/5/commits/c0e5c22a05d73b13cad58dd793db5d58a525e1f9)

File: .github/wiki/Getting-Started.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (hoangmirs) [reply .github/wiki/Getting-Started.md:20]: Feel free to create issues in the `rails-templates` so that the team can know and update it 💪

**Review by olivierobert — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE/RELEASE_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+Link to the milestone on Github e.g. https://github.com/nimblehq/ic-rails-huey-toby/milestone/41?closed=1
```
Comment (hoangmirs) [inline .github/PULL_REQUEST_TEMPLATE/RELEASE_TEMPLATE.md:1]: I think we just need the URL here:
```suggestion
https://github.com/nimblehq/ic-rails-huey-toby/milestone/{milestone-id}?closed=1
```

**Review by hoangmirs — APPROVED** [review]

**Review by Tuubz — APPROVED** [review]
💪

**Pull request merged**